airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -527,6 +523,12 @@ public KubePodProcess(final boolean isOrchestrator,
     }, 20, TimeUnit.MINUTES);
     DogStatsDMetricSingleton.recordTimeGlobal(MetricsRegistry.KUBE_POD_PROCESS_CREATE_TIME_MILLISECS, System.currentTimeMillis() - start);
 
+    exitCodeFuture = new CompletableFuture<>();
+
+    podWatch = fabricClient.resource(podDefinition).watch(new ExitCodeWatcher(
```

Comment:
do we need to kill the watcher?

Reply:
nice! missed that. can we also make sure to update the class's docstring to it's easy to digest this? (I know this is still WIP, but want to make sure we don't forget)